packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,373 @@
+          src={network.logo}
+          alt={`${network.name} logo`}
+          className="h-full w-auto"
+          width={24}
+          height={24}
+        />
+      </span>
+      <div className={twMerge('flex flex-col items-start gap-1')}>
+        <span className="truncate leading-none">{network.name}</span>
+        {description && (
+          <p className="whitespace-pre-wrap text-left text-xs leading-[1.15]">
+            {description}
+          </p>
+        )}
+        {chainType && (
+          <p className="text-[10px] leading-none">
+            {chainType} Chain, {nativeTokenData?.symbol ?? 'ETH'} is the native
+            gas token
+          </p>
+        )}
+      </div>
+    </button>
+  )
+}
+
+function AddCustomOrbitChainButton() {
+  const [, setQueryParams] = useArbQueryParams()
+  const { isTestnetMode } = useIsTestnetMode()
+
+  if (!isTestnetMode) {
+    return null
+  }
+
+  return (
+    <button
+      className="text-sm underline"
+      onClick={() => setQueryParams({ settingsOpen: true })}
+    >
+      <span>Add Custom Orbit Chain</span>
+    </button>
+  )
+}
+
+function NetworksPanel({
+  networks,
+  onNetworkRowClick,
+  close
+}: {
+  networks: ChainId[]
+  onNetworkRowClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const [errorMessage, setErrorMessage] = useState('')
+  const [networkSearched, setNetworkSearched] = useState('')
+  const [debouncedNetworkSearched, setDebouncedNetworkSearched] = useState('')
+  const listRef = useRef<List>(null)
+  const { isTestnetMode } = useIsTestnetMode()
+
+  const testnetToggleClassNames = {
+    switch:
+      'ui-checked:bg-black/20 ui-not-checked:bg-black/20 [&_span]:ui-not-checked:bg-black'
+  }
+
+  useEffect(() => {
+    listRef.current?.recomputeRowHeights()
+  }, [isTestnetMode])
+
+  useDebounce(
+    () => {
+      setDebouncedNetworkSearched(networkSearched)
+    },
+    200,
+    [networkSearched]
+  )
+
+  function getRowHeight({ index }: { index: number }) {
+    const rowItemOrChainId = networksToShowWithChainTypeInfo[index]
+    if (!rowItemOrChainId) {
+      return 0
+    }
+    if (typeof rowItemOrChainId === 'object') {
+      return 65
+    }
+    const rowItem = getBridgeUiConfigForChain(rowItemOrChainId)
+    if (rowItem.description) {
+      return 90
+    }
+    return 52
+  }
+
+  const networksToShowWithChainTypeInfo = useMemo(() => {
+    const _networkSearched = debouncedNetworkSearched.trim().toLowerCase()
+
+    if (_networkSearched) {
+      return networks.filter(network => {
+        const networkName =
+          getBridgeUiConfigForChain(network).network.name.toLowerCase()
+        return networkName.includes(_networkSearched)
+      })
+    }
+
+    const coreNetworks = networks.filter(
+      chainId => !isNetwork(chainId).isOrbitChain
+    )
+    const orbitNetworks = networks.filter(
+      chainId => isNetwork(chainId).isOrbitChain
+    )
+
+    return [
+      chainGroupInfo.core,
```

Comment:
I find it a bit odd that headers and values are in the same array and all are rendered in `rowRenderer`.

It should be cleaner if we stored only `coreNetworks` and `orbitNetwork` in an array, possibly as a matrix. Then we could map through it to render different Lists, and render headers above the list rows.

How does that sound?

Reply:
Or not even a matrix, just 2 different arrays and then 2 different lists

Reply:
two lists would introduce two separate scroll bars as they are separate lists so they have to be on the same list

and we need the full list of rows to get the dynamic row height for the category rows